src/ophyd_async/tango/demo/_counter.py

```diff
@@ -0,0 +1,50 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    AsyncStatus,
+    ConfigSignal,
+    HintedSignal,
+    Signal,
+    SignalX,
+)
+from ophyd_async.tango import TangoReadable
+
+
+@dataclass
+class TangoCounterConfig:
+    sample_time: Optional[float] = None
+
+
+class TangoCounter(TangoReadable):
+    # Enter the name and type of the signals you want to use
+    # If type is None or Signal, the type will be inferred from the Tango device
+    counts: Signal
+    sample_time: Signal
+    state: Signal
+    reset: Signal
+    start: SignalX
```

Comment:
I think the requirement for type hints is a strength of ophyd-async in general, where we have control systems that are loosely defined, being able to say `counts = Signal[float]` and knowing I will get a sensible error message at connect time if the control system does not actually return a float is very useful. I wouldn't want to see that requirement lost here. See also my other comments about maintaining consistency with the rest of the codebase.